server/src/test/java/org/elasticsearch/cluster/routing/allocation/ShardsHealthIndicatorServiceTests.java

```diff
@@ -0,0 +1,310 @@
+
+        assertThat(
+            service.calculate(),
+            equalTo(
+                createExpectedResult(
+                    YELLOW,
+                    String.format(
+                        Locale.ROOT,
+                        "This cluster has %d shards including %d primaries (%s unreplicated) and %d replicas.",
+                        greenIndices.size() * 2 + 1,
+                        greenIndices.size() + 1,
+                        yellowIndex.shards().get(1).shardId().toString(),
+                        greenIndices.size()
+                    ),
+                    createDetails(
+                        appendToCopy(greenIndices, yellowIndex),
+                        greenIndices,
+                        List.of(yellowIndex),
+                        List.of(),
+                        List.of(),
+                        List.of()
+                    )
+                )
+            )
+        );
+    }
+
+    public void testShouldBeRedWhenNotActive() {
+        var greenIndices = randomList(1, 10, indexGenerator("green-index-", STARTED, STARTED));
+        var redIndex = index("red-index", UNASSIGNED);
+        var clusterState = createClusterStateWith(appendToCopy(greenIndices, redIndex));
+        var service = createAllocationHealthIndicatorService(clusterState);
+
+        assertThat(
+            service.calculate(),
+            equalTo(
+                createExpectedResult(
+                    RED,
+                    String.format(
+                        Locale.ROOT,
+                        "This cluster has %d shards including %d primaries (%s unreplicated) (%s unallocated) and %d replicas.",
+                        greenIndices.size() * 2 + 1,
+                        greenIndices.size() + 1,
+                        redIndex.shards().get(1).shardId().toString(),
+                        redIndex.shards().get(1).shardId().toString(),
+                        greenIndices.size()
+                    ),
+                    createDetails(greenIndices, greenIndices, List.of(redIndex), List.of(), List.of(), List.of(redIndex))
+                )
+            )
+        );
+    }
+
+    public void testShouldBeGreenIfReplicaUnavailableDueToRestarting() {
+        var greenIndices = randomList(1, 10, indexGenerator("green-index-", STARTED, STARTED));
+        var restartingIndex = index("restarting-index", STARTED, UNASSIGNED_RESTARTING);
+        var clusterState = createClusterStateWith(appendToCopy(greenIndices, restartingIndex));
+        var service = createAllocationHealthIndicatorService(clusterState);
+
+        assertThat(
+            service.calculate(),
+            equalTo(
+                createExpectedResult(
+                    GREEN,
+                    String.format(
+                        Locale.ROOT,
+                        "This cluster has %d shards including %d primaries and %d replicas (%s temporary unallocated due to node restarting).",
```

Comment:
Should we explicitly highlight shards that are temporary unavailable due to node restart (even though they are not affecting health status) or just silently ignore them?